tests/cairo_files/instructions/test_system_operations.cairo

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: MIT
+
+%lang starknet
+
+// Starkware dependencies
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.cairo_builtins import HashBuiltin, BitwiseBuiltin
+from starkware.cairo.common.uint256 import Uint256
+
+// Local dependencies
+from kakarot.model import model
+from kakarot.stack import Stack
+from kakarot.execution_context import ExecutionContext
+from kakarot.instructions.memory_operations import MemoryOperations
+from kakarot.instructions.system_operations import SystemOperations
+from kakarot.constants import Constants
+
+@view
+func __setup__{
+    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
+}() {
+    return ();
+}
+
+func init_context{
+    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
+}() -> model.ExecutionContext* {
+    alloc_locals;
+    let (bytecode) = alloc();
+    assert [bytecode] = 00;
+    tempvar bytecode_len = 1;
+    let (calldata) = alloc();
+    assert [calldata] = '';
+    local call_context: model.CallContext* = new model.CallContext(
+        bytecode=bytecode, bytecode_len=bytecode_len, calldata=calldata, calldata_len=1, value=0
+        );
+    let ctx: model.ExecutionContext* = ExecutionContext.init(call_context);
+    return ctx;
+}
+
+@external
+func test_exec_revert{
+    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
+}(reason: felt) {
+    // Given
+    alloc_locals;
+    let ctx: model.ExecutionContext* = init_context();
+    let stack: model.Stack* = Stack.init();
+    let stack: model.Stack* = Stack.push(stack, Uint256(reason, 0));
+    let stack: model.Stack* = Stack.push(stack, Uint256(0, 0));
+    let ctx: model.ExecutionContext* = ExecutionContext.update_stack(ctx, stack);
+
+    // When
+    let ctx: model.ExecutionContext* = MemoryOperations.exec_mstore(ctx);
+
+    // Then
+    let stack: model.Stack* = Stack.push(ctx.stack, Uint256(32, 0));
+    let stack: model.Stack* = Stack.push(stack, Uint256(0, 0));
+    let ctx: model.ExecutionContext* = ExecutionContext.update_stack(ctx, stack);
+    SystemOperations.exec_revert(ctx);
```

Comment:
don't we miss an `assert` on the calling context return data?

Reply:
I'm expecting the test to revert as it uses StarkNet's native revert by using pytest's XFAIL. Not sure if updating the return data and asserting it would be possible